### PR TITLE
Add Wallet Service API to create change addresses

### PR DESCRIPTION
**Pull request by troggy**
I need such an API to construct transactions for Colu tokens

### CODE REVIEW

**Comment by dabura667** [comment]
This opens up a can of worms that could break BIP44 compatibility.

You need to add methods to routinely check change addresses now. Currently BWS is assuming change addresses are used in order.

**Comment by troggy** [comment]
You are right, I missed the point of ``_canCreateAddress``. I will make necessary changes and update PR

**Comment by troggy** [comment]
I've adopted ``_canCreateAddress`` to work for change addresses also, so now it should keep track of the change address order.

File: test/integration/server.js [hunk 1]
```diff
@@ -1155,75 +1155,103 @@ describe('Wallet service', function() {
...
-        Defaults.MAX_MAIN_ADDRESS_GAP = 2;
-        helpers.stubAddressActivity([]);
-        async.map(_.range(2), function(i, next) {
-          server.createAddress({}, next);
-        }, function(err, addresses) {
-          addresses.length.should.equal(2);
+      [
+        { isChange: false, branch: 0, thirdAddress: '1GdXraZ1gtoVAvBh49D4hK9xLm6SKgesoE'},
+        { isChange: true, branch: 1, thirdAddress: '13ee82QuHiEz7y5rZjPTsZKfox2rfYeGFv'},
+      ].forEach(function(param) {
```
Comment (troggy) [inline test/integration/server.js]: this way we can avoid lots of copy and paste. Such a way of test parameterization looks kind of ugly, feel free to suggest yours

**Comment by troggy** [comment]
I've reworked this one. Now there is a separate endpoint for change addresses. If returned address is not used, then next call will return the same address.

**Comment by troggy** [comment]
@dabura667 Thanks for the feedback on this PR. Any chance you can have look at the updated version?

File: lib/server.js [hunk 2]
```diff
@@ -853,6 +835,91 @@ WalletService.prototype.getMainAddresses = function(opts, cb) {
...
+ */
+ WalletService.prototype.getNextChangeAddress = function(cb) {
+   var self = this;
+
+   self._runLocked(cb, function(cb) {
+     self.getWallet({}, function(err, wallet) {
+       if (err) return cb(err);
+       if (!wallet.isComplete()) return cb(Errors.WALLET_NOT_COMPLETE);
+       return self._getChangeAddress(wallet, function(err, address) {
+         return self.storage.storeAddress(address, function(err) {
```
Comment (matiu) [inline lib/server.js]: storeAddress is called always, even if the address is not new.

Comment (rafiq34) [reply lib/server.js]: I am use bitcoin address. Please. Active  
1K9312KKDshshNy43GFwCnkw44zeF4pWW6

**Comment by Keensyst** [comment]
I can't receive any Btc , I don't know what the problem is

**Comment by jessadix34** [comment]
git@github.com:jessadix34/sw-precache-background..git

**Comment by jessadix34** [comment]
F98 @ @dabura667 

1.

**Comment by AUTHAI-bit** [comment]
I don't know problems Or have a mistake

**Pull request closed without merging**